api/src/services/import-services/marking/marking-header-configs.ts

```diff
@@ -0,0 +1,237 @@
+import { z } from 'zod';
+import { CSVConfigUtils } from '../../../utils/csv-utils/csv-config-utils';
+import { CSVCellValidator, CSVError, CSVParams } from '../../../utils/csv-utils/csv-config-validation.interface';
+import { validateZodCell } from '../../../utils/csv-utils/csv-header-configs';
+import { NestedRecord } from '../../../utils/nested-record';
+import { ICritterDetailed } from '../../critterbase-service';
+import { findCapturesFromDateTime } from '../utils/datetime';
+import { MarkingCSVStaticHeader } from './import-markings-service';
+
+/**
+ * Get the marking identifier cell validator.
+ *
+ * Rules:
+ *  1. The cell can be a string with a length between 1 and 50
+ *  2. The cell can be a number with a min value of 0
+ *  3. The cell can be optional
+ *
+ * @returns {*} {CSVCellValidator} The validate cell callback
+ */
+export const getMarkingIdentifierCellValidator = (): CSVCellValidator => {
+  return (params: CSVParams) => {
+    return validateZodCell(params, z.union([z.string().trim().min(1).max(50), z.number().min(0)]).optional());
+  };
+};
+
+/**
+ * Get the marking alias cell validator.
+ *
+ * Note: Modifies the mutateCell value to the `critter_id`
+ *
+ * Rules:
+ *  1. The alias must exist in the surveyAliasMap ie: critter alias -> critter
+ *  2. The alias (critter) must have Critterbase captures
+ *
+ *  @param {Map<string, ICritterDetailed>} surveyAliasMap The survey alias map
+ *  @returns {*} {CSVCellValidator} The validate cell callback
+ */
+export const getMarkingAliasCellValidator = (surveyAliasMap: Map<string, ICritterDetailed>): CSVCellValidator => {
+  return (params: CSVParams) => {
+    if (params.cell === undefined) {
+      return [];
+    }
+
+    const critter = surveyAliasMap.get(String(params.cell).toLowerCase());
+
+    if (!critter) {
+      return [
+        {
+          error: `Unable to find a matching survey critter`,
+          solution: `Use a valid critter alias that exists in the Survey`
+        }
+      ];
+    }
+
+    // If the critter has no captures
+    if (critter.captures.length === 0) {
+      return [
+        {
+          error: `Animal has no captures`,
+          solution: `Add captures to animal`
+        }
+      ];
+    }
+
+    // Set the critter id in the state for the setter
+    params.mutateCell = critter.critter_id;
+
+    return [];
+  };
+};
+
+/**
+ * Get the marking type cell validator.
+ *
+ * Rules:
+ *  1. The cell must be a valid marking type ie: exists in the markingTypes set
+ *
+ * @param {Set<string>} markingTypes The marking types
+ * @returns {*} {CSVCellValidator} The validate cell callback
+ */
+export const getMarkingTypeCellValidator = (markingTypes: Set<string>): CSVCellValidator => {
+  return (params: CSVParams) => {
+    if (params.cell === undefined) {
+      return [];
+    }
+
+    if (!markingTypes.has(String(params.cell).toLowerCase())) {
```

Comment:
In the parent, the markings are transformed to lowercase. Wondering if that should be done at this level instead, so the caller doesn't have to know what case the values need to be in.

Reply:
Though, I guess that is less performant since it would be done every time then. In that case, maybe just add a jsdoc comment that the values will be compared in lowercase and must be transformed

Reply:
Same for the other string based validators

Reply:
Yea the pattern I have been using is just casting everything to lowercase. But to your point, it's kinda confusing for the caller to have to know to case everything to lowercase. I can make a change for this.

Reply:
Updated